### ISSUE TITLE
Comma in indexing expression produces incorrect error location and suggestion to add a semicolon

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust#161147]
### Code

```Rust
fn example() {
    (foo[
        bar
        .baz(),
    ])
}
```

### Current output

```Shell
error: expected `;`, found `[`
 --> src/lib.rs:2:9
  |
2 |     (foo[
  |         ^
  |
help: consider adding `;` here
  |
2 |     (foo;[
  |         +
```

### Desired output

```Shell
error: expected `]`, found `,`
  |
4 |         .baz(),
  |               ^
```

### Rationale and extra context

The existing error message points to the wrong line, and suggests that a semicolon could help, when in fact it would make things worse.

### Other cases

This problem is sensitive to line breaks — it does not happen if the extra comma is on the same line as `foo`.

### Rust Version

```Shell
rustc 1.97.1 (8bab26f4f 2026-07-14)
binary: rustc
commit-hash: 8bab26f4f68e0e26f0bb7960be334d5b520ea452
commit-date: 2026-07-14
host: aarch64-apple-darwin
release: 1.97.1
LLVM version: 22.1.6
```

**Comment by hu-jeff** [comment]
Hey, I'm Jeff! I'd like to take a look at fixing this issue as my first contribution to rust :D

**Comment by kpreid** [comment]
@hu-jeff Welcome! I would guess that the “expected semicolon” claim is due to incorrectly implemented *recovery* (attempting to re-parse the input using a guess about what the user meant to write), but I haven’t looked into whether this is true.
